src/components/Clock/Clock.tsx

```diff
@@ -0,0 +1,80 @@
+import { Component } from 'react';
+
+function getTime(): string {
+  const time = new Date();
+
+  return time.toUTCString().slice(-12, -4);
+}
+
+type Props = {
+  name: string;
+};
+
+type State = {
+  time: string;
+  clockName: string;
+};
+
+export class Clock extends Component<Props, State> {
+  state = {
+    time: getTime(),
+    clockName: this.props.name,
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const time = getTime();
+
+      this.setState({ time });
+
+      // eslint-disable-next-line no-console
+      console.info(time);
+    }, 1000);
+    this.setState({ clockName: this.props.name });
+  }
```

Comment:
The clockName should update every 3300ms, but it's not updating automatically. You should set an interval for updating the clockName.

Reply:
> The clockName should update every 3300ms, but it's not updating automatically. You should set an interval for updating the clockName.

Hello. New clock name generates in App, and transfer to a Clock component during props. In componentDidUpdate  - state of Clock component updates and renders.